Model/Order/PlaceOrder.php

```diff
@@ -137,11 +181,234 @@ public function place(string $shopId, OrderDataInterface $order): ResultInterfac
+                         *     severity: string,
+                         *     sub_type: string,
+                         *     code?: string,
+                         *     transactions?: array{
+                         *         gateway: string,
+                         *         gateway_id: string,
+                         *         amount: int,
+                         *         transaction_id: string,
+                         *         reference_transaction_id: string|null,
+                         *         response_code: string,
+                         *         status: 'success'|'failure'
+                         *     }[]
+                         * } $error
+                         */
+                        function (array $error): ErrorInterface {
+                            return $this->errorFactory->create(
+                                [
+                                    'code' => (int)($error['code'] ?? 0),
+                                    'type' => $error['type'],
+                                    'message' => $error['message']
+                                ]
+                            );
+                        },
+                        $authorizedPayments['errors']
+                    )
+                ]
+            );
+        }
+
+        if (!array_key_exists('data', $authorizedPayments) || count($authorizedPayments['data']) === 0) {
+            return $this->responseFactory->create(
+                [
+                    'errors' => [
+                        $this->errorFactory->create(
+                            [
+                                'message' => __(
+                                    'There are no authorized payments available for order "%1"',
+                                    $publicOrderId
+                                ),
+                                'code' => 500,
+                                'type' => 'server.bold_checkout_api_error'
+                            ]
+                        )
+                    ]
+                ]
+            );
+        }
+
+        $transactions = array_filter(
+            $authorizedPayments['data']['transactions'],
+            /**
+             * @param array{
+             *     gateway: string,
+             *     payment_id: string,
+             *     amount: int,
+             *     transaction_id: string,
+             *     currency: string,
+             *     step: string,
+             *     status: 'success'|'failure'|'',
+             *     tender_type: string,
+             *     tender_details: array{
+             *         brand: string,
+             *         last_four: string,
+             *         bin: string,
+             *         expiration: string
+             *     },
+             *     gateway_response_data: string[]
+             * } $transaction
+             *
+             * TODO: verify that the status used for filtering is correct
```

Comment:
[Clarification] Has this been done?

Reply:
No, that will require testing of the error cases (use one of the credit card numbers that results in a failed payment).

Reply:
Fixed.